test/unit/controller/ResultControllerTest.php

```diff
@@ -0,0 +1,307 @@
+
+        /** @var ServerRequestInterface|MockObject $requestMock */
+        $requestMock = $this->createMock(ServerRequestInterface::class);
+        $requestMock->method('getMethod')->willReturn('POST');
+
+        /** @var ResponseInterface|MockObject $responseMock */
+        $responseMock = $this->createMock(ResponseInterface::class);
+        $responseMock->method('withStatus')->with(500)->willReturn($responseMock);
+        $responseMock->method('withHeader')->with('Content-Type', 'application/xml')->willReturn($responseMock);
+        $responseMock->method('withBody')
+            ->with($this->callback(function (StreamInterface $stream) {
+                return 'xmlout' === (string) $stream;
+            }))
+            ->willReturn($responseMock);
+
+        /** @var BasicResponseSerializer|MockObject $ltiResultServiceMock */
+        $basicResponseSerializerMock = $this->createMock(BasicResponseSerializer::class);
+        $basicResponseSerializerMock->expects($this->once())
+            ->method('toXml')
+            ->with($this->callback(function (LisOutcomeResponseInterface $response) {
+                return
+                    $response->getStatus() === LisOutcomeResponseInterface::STATUS_INTERNAL_ERROR &&
+                    $response->getCodeMajor() === LisOutcomeResponseInterface::CODE_MAJOR_FAILURE;
+            }))
+            ->willReturn('xmlout');
+
+        /** @var LoggerInterface|MockObject $loggerMock */
+        $loggerMock = $this->createMock(LoggerInterface::class);
+        $loggerMock->expects($this->once())->method('error');
+
+        $controller = new ResultController();
+        $controller->setServiceLocator($this->getServiceLocatorMock([
+            LisAuthAdapterFactory::class => $lisAuthAdapterFactory,
+            BasicResponseSerializer::class => $basicResponseSerializerMock
+        ]));
+        $controller->setLogger($loggerMock);
+        $controller->setRequest($requestMock);
+        $controller->setResponse($responseMock);
+        $controller->manageResults();
+        $response = $controller->getPsrResponse();
+        $this->assertSame($responseMock, $response);
+    }
+
+    /**
+     * @throws common_exception_MethodNotAllowed
+     * @throws tao_models_classes_UserException
+     */
+    public function testManageResult()
+    {
+        /** @var LisOutcomeRequest|MockObject $lisRequestMock */
+        $lisRequestMock = $this->createMock(LisOutcomeRequest::class);
+
+        /** @var LisOutcomeRequestParser|MockObject $lisOutcomeRequestParserMock */
+        $lisOutcomeRequestParserMock = $this->createMock(LisOutcomeRequestParser::class);
+        $lisOutcomeRequestParserMock->expects($this->once())
+            ->method('parse')
+            ->with('request_body')
+            ->willReturn($lisRequestMock);
+
+        /** @var TenantLtiProvider|MockObject $tenantLtiProviderMock */
+        $tenantLtiProviderMock = $this->createMock(TenantLtiProvider::class);
```

Comment:
Looks like you forgot to move this to another extension with all other tenant related code.

Reply:
You are absolutely right, fixed